test/Safe.test.js

```diff
@@ -75,6 +75,59 @@ describe("ERC20Safe", async function () {
     });
   });
 
+  describe("ERC20Safe - setting batch settle limit works as expected", async function () {
+    it("is default correct", async function () {
+      const tenMinutes = 40;
+      expect(await safe.batchSettleLimit()).to.eq(tenMinutes);
+    });
+    it("reverts", async function () {
+      await expect(safe.connect(adminWallet).setBatchSettleLimit(50)).to.be.revertedWith("Pausable: not paused");
+
+      await safe.pause();
+
+      await expect(safe.connect(otherWallet).setBatchSettleLimit(50)).to.be.revertedWith(
+        "Access Control: sender is not Admin",
+      );
+
+      // Having no batch
+      await expect(safe.connect(adminWallet).setBatchSettleLimit(30)).to.be.revertedWith(
+        "Cannot decrease batchSettleLimit under the value of batch block limit",
+      );
+
+      // Creating a batch
+      await safe.whitelistToken(genericERC20.address, defaultMinAmount, defaultMaxAmount);
+      await genericERC20.approve(safe.address, "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
+      await genericERC20.mint(adminWallet.address, "1000000");
+      await safe.unpause();
+      await safe.deposit(
+        genericERC20.address,
+        defaultMinAmount,
+        Buffer.from("c0f0058cea88a2bc1240b60361efb965957038d05f916c42b3f23a2c38ced81e", "hex"),
+      );
+
+      await network.provider.send("evm_increaseTime", [3600]);
+      for (let i = 0; i < 41; i++) {
+        await network.provider.send("evm_mine");
+      }
+      await safe.pause();
+      await expect(safe.connect(adminWallet).setBatchSettleLimit(30)).to.be.revertedWith(
+        "Cannot decrease batchSettleLimit under the value of batch block limit",
+      );
+
+      await safe.unpause();
```

Comment:
here you can also check that the transaction fails with "Cannot change batchSettleLimit with pending batches"

This should be test also if _shouldCreateNewBatch() is true but  _isBatchFinal returns false.